### PR TITLE
Added doc about function arguments

### PR DESCRIPTION
**Pull request by robhuzzey**
#### What does this PR do? (please provide any background)

Adds a line about function arguments when coding & making them clearer.
#### What tests does this PR have?

N/A
#### How can this be tested?

Read it... nod & +1... or comment.
#### Screenshots / Screencast

N/A
#### What gif best describes how you feel about this work?

nope

---
- [x] I have checked our general [contributing document](https://github.com/holidayextras/culture/blob/master/CONTRIBUTING.md) and the project specific [contributing document](../blob/master/CONTRIBUTING.md) (if present) and I'm happy for this to be reviewed.
#### Reviewers

**Review 1**
- [ ] :+1:

**Review 2** *
- [x] :+1:

**Review 3** _(optional)_
- [ ] :+1:

By adding a +1 you are confirming you have...
- Witnessed the work behaving as expected (this could be on the author's machine or screencast).
- Checked for coding anti-patterns.
- Checked for appropriate test coverage.
- Checked all the tests are passing.

\*  for HX this review must be completed by an SE, SA or Project Guru

### CODE REVIEW

**Comment by djbeaumont** [comment]
:+1:

File: coding-principles.md [hunk 1]
```diff
@@ -31,6 +31,27 @@ var myUniqueArray = _.uniq(myArray);
...
+  sku: 1234,
+  price: 201.99,
+  quantity: 1,
+  position: 4,
+  type: 'Hotel',
+  page: '/hotels',
+  isPromoted: true
+});
+```
+
```
Comment (kevinhodges) [inline coding-principles.md:54]: to provide a devils advocate here, you haven't demonstrated the power of many arguments, consider how the function definition would look, your bad becomes the good and vice versa.  Happy with either way myself, just wanted to provide some thoughts.

Good:

``` javascript
function log( sku, price, quantity, position, type, page, isPromoted ) {
    // all the things
}
```

Bad:

``` javascript
function log( object ) {
    // all the things
}

```

Comment (msaspence) [inline coding-principles.md:54]: Comments can help clear this up in your bad example

If there is a default set of arguments that the object is merged into, inside the bad example, that usually makes the code "self documenting". And each value can have a comment if necessary.

File: coding-principles.md [hunk 2]
```diff
@@ -31,6 +31,27 @@ var myUniqueArray = _.uniq(myArray);
 
 One might argue that extensive use of this syntactic sugar will lead to a slower codebase, but modern JavaScript engines do a great job of optimising the code we write to the point that the readibility benefits outweigh any minor performance quibbles. In browsers especially, a huge amount more CPU time is lost to rendering than actual application logic.
 
+## Keep arguments to functions simple
+Only pass a few to each function. If you find there are too many, re-consider the functions purpose or pass an object. The goal is to keep it readable:
```
Comment (hx-markterry) [inline coding-principles.md]: `functions` should be `function's`

**Comment by pmcnr-hx** [comment]
I would reinforce the fact that too many arguments is more usually than not a code smell indicating that the function should be refactored into smaller functions. Also it doesn't need to be an all or nothing for replacing with a single object parameter: you may want to keep some of the parameters as standalone while grouping the logically related ones in an object.

File: coding-principles.md (hunk 1 above)
Comment (robhuzzey) [inline coding-principles.md:54]: Yes, that's a great point. I was thinking that's how we'd use it but the example doesn't illustrate this. I will update, thanks.

**Comment by robhuzzey** [comment]
Thanks @pmcnr-hx @msaspence @hx-markterry @kevinhodges I've updated a little more, please can you feedback or :+1: :)

**Comment by hx-markterry** [comment]
ok with me :slot_machine:

File: coding-principles.md [hunk 3]
```diff
@@ -31,6 +31,63 @@ var myUniqueArray = _.uniq(myArray);
...
+  
+  // Check `requirements` against `options` passed in & handle accordingly.
+
+  var defaults = {
+    isPromoted: false
+  };
+  
+  // Merge `options` with `defaults` here
+  ... etc
+}
```
Comment (msaspence) [inline coding-principles.md:69]: These examples feel like logic expressed through a strange combo of a. JavaScript, b. Comments with // and c. a comment without //

I don't fell like the comments in the code really add much that the paragraph above hasn't already laid out, and that full examples in JavaScript would add value here.

Comment (robhuzzey) [inline coding-principles.md:69]: @msaspence I wasn't too sure about confusing the point with full examples as if this were a "real" piece of code, you could use `lodash` to check props etc.

To be honest I struggled with what is a "good" function body for this, really the point is about discouraging function declarations that have so many arguments that it becomes hard to interpret the functions purpose.

**Pull request closed without merging**